### PR TITLE
Smoke test for the R Test Explorer

### PR DESCRIPTION
**Pull request by testlabauto**
Smoke test for the R Test Explorer.

Uses an R package embedded in [qa-example-content](https://github.com/posit-dev/qa-example-content/tree/main/workspaces/r_testing).

Since qa-example-content is already downloaded for the smoke tests, the test just needs to change directory to the embedded package for Positron to identify the current folder as an R package.  After navigating to the embedded package, all tests are run and their expected outcomes are checked.

### QA Notes

All smoke tests should pass.

### CODE REVIEW

File: test/automation/src/positron/positronQuickInput.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *  Extends Microsoft's QuickInput functionality to provide Positron-specific functionality.
+ */
+export class PositronQuickInput {
+
+	constructor(private code: Code) { }
+
+	async clickOkOnQuickInput(): Promise<void> {
+		await this.code.driver.getLocator(QUICKINPUT_OK_BUTTON).click();
+	}
+}
```
[the remark this answers is not in the document]
Comment (jonvanausdeln) [reply test/automation/src/positron/positronQuickInput.ts:21]: I'm fine either way. But if we pull it out, I'd advocate for a separate Positron class that extends the base, to avoid confusion when merging upstream. And then we could always just the new Positron one. (we could also just make that a separate issue/PR)

**Review by jonvanausdeln — DISMISSED** [review]
LGTM!

File: test/automation/src/positron/positronTestExplorer.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+const COMPUTED_STATE = '.computed-state';
+const TEST_EXPLORER_ICON = '.codicon-test-view-icon';
+const RUN_ALL = '.codicon-testing-run-all-icon';
+
+/*
+ *  Reuseable Positron test explorer functionality for tests to leverage.
+ */
+export class PositronTestExplorer {
+
+	constructor(private code: Code, private positronExplorer: PositronExplorer) { }
```
[the remark this answers is not in the document]
Comment (testlabauto) [reply test/automation/src/positron/positronTestExplorer.ts:21]: Good question.  My only hesitation was the fact that the Test Explorer had a filter (search) whereas I didn't see the same feature in regular Explorer.  I could go either way here.

File: test/automation/src/positron/positronQuickInput.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (testlabauto) [reply test/automation/src/positron/positronQuickInput.ts:21]: We have been trying not to modify the Microsoft base test automation classes, but perhaps that was not the best choice.  I am not 100% sure why we leaned towards this approach.  I think I originally went with @seeM's opinion on this, but again, I could go either way.  Whatever we do with QuickInput we should also do with our PositronNotebooks.ts class.

[the remark this answers is not in the document]
Comment (seeM) [reply test/automation/src/positron/positronQuickInput.ts:21]: Interesting, I don't recall having an opinion on this!

If we're only adding one method, I don't see an issue with updating the `QuickInput` class. I actually did that myself when adding the `selectQuickInputElementContaining` method.

[the remark this answers is not in the document]
Comment (testlabauto) [reply test/automation/src/positron/positronQuickInput.ts:21]: I must have misremembered.  I will add the function to the existing MS class.

File: test/automation/src/positron/positronTestExplorer.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (testlabauto) [reply test/automation/src/positron/positronTestExplorer.ts]: Extended PositronExplorer instead!

File: test/smoke/src/areas/positron/test-explorer/test-explorer.test.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+					await PositronRFixtures.SetupFixtures(this.app as Application);
+
+					userSettings = new PositronUserSettingsFixtures(app);
+
+					// don't use native file picker
+					await userSettings.setUserSetting([
+						'files.simpleDialog.enable',
+						'true',
+					]);
```
Comment (jennybc) [inline test/smoke/src/areas/positron/test-explorer/test-explorer.test.ts:34]: I'm just curious. What's this about? From the outside, it would seem unrelated.

**Review by jennybc — APPROVED** [review]
As a non-expert on the tests, I can't add a whole lot here. But I will say that this seems like a reasonable smoke test, in terms of tickling the test explorer and making sure the lights come on.

File: test/smoke/src/areas/positron/test-explorer/test-explorer.test.ts (hunk 3 above)
Comment (testlabauto) [reply test/smoke/src/areas/positron/test-explorer/test-explorer.test.ts:34]: Good question.  That eliminates the native file picker dialog so the test is not OS dependent.  Once that setting is set, a VSCode file picker is available instead.

Comment (jennybc) [reply test/smoke/src/areas/positron/test-explorer/test-explorer.test.ts:34]: And file picking comes up ... when opening the folder that is a package that we use as the test fixture?

Comment (testlabauto) [reply test/smoke/src/areas/positron/test-explorer/test-explorer.test.ts:34]: Yeah, exactly.  The test opens with the qa-example-content directory, and then I need to navigate down to the subdirectory to make Positron think its an R package.
